### PR TITLE
Add Integration tests for Custom Text Management API

### PR DESCRIPTION
**Pull request by UdeshAthukorala**
## Purpose

Add Integration tests for the Custom-Text Management API.

##TODO

- [x] Bump framework version
- [x] Bump branding-prefernce-management version
- [x] Bump api-server version

### CODE REVIEW

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/rest/api/server/branding/preference/management/v1/BrandingPreferenceManagementFailureTest.java [hunk 1]
```diff
@@ -167,4 +167,91 @@ public void testDeleteNotExistingBrandingPreference() {
...
+    @Test
+    public void testUpdateNotExistingCustomTextPreference() throws IOException {
+
+        String body = readResource("update-custom-text.json");
+        Response response = getResponseOfPut(CUSTOM_TEXT_API_BASE_PATH, body);
+        validateErrorResponse(response, HttpStatus.SC_NOT_FOUND, "BPM-60006");
+    }
+
+    @Test
+    public void testUpdateCustomTextPreferenceWithEmptyJsonPreference() throws IOException {
```
Comment (ThaminduDilshan) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/rest/api/server/branding/preference/management/v1/BrandingPreferenceManagementFailureTest.java:223]: Can combine this test and `testAddCustomTextPreferenceConflict` with a data provider

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/rest/api/server/branding/preference/management/v1/BrandingPreferenceManagementSuccessTest.java [hunk 2]
```diff
@@ -318,4 +318,126 @@ public void testDeleteBrandingPreferenceByLocaleQueryParam() throws IOException
...
+
+        response.then()
+                .log().ifValidationFails()
+                .assertThat()
+                .statusCode(HttpStatus.SC_CREATED)
+                .header(HttpHeaders.LOCATION, notNullValue());
+        String location = response.getHeader(HttpHeaders.LOCATION);
+        assertNotNull(location);
+
+        JSONObject expectedPreference = new JSONObject(new JSONObject(readResource("add-custom-text.json")).
```
Comment (ThaminduDilshan) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/rest/api/server/branding/preference/management/v1/BrandingPreferenceManagementSuccessTest.java:336]:
```suggestion
        JSONObject expectedPreference = new JSONObject(new JSONObject(body).
```

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/rest/api/server/branding/preference/management/v1/BrandingPreferenceManagementSuccessTest.java [hunk 3]
```diff
@@ -318,4 +318,126 @@ public void testDeleteBrandingPreferenceByLocaleQueryParam() throws IOException
...
+                get("preference").toString());
+        JSONObject receivedPreference = new JSONObject(new JSONObject(response.asString()).
+                get("preference").toString());
+        Assert.assertTrue(areJSONObjectsEqual(expectedPreference, receivedPreference),
+                "The custom text preference schema of the Response is incorrect");
```
Comment (ThaminduDilshan) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/rest/api/server/branding/preference/management/v1/BrandingPreferenceManagementSuccessTest.java:341]:
```suggestion
                "The custom text preference schema of the Response is incorrect.");
```

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/rest/api/server/branding/preference/management/v1/BrandingPreferenceManagementSuccessTest.java [hunk 4]
```diff
@@ -318,4 +318,126 @@ public void testDeleteBrandingPreferenceByLocaleQueryParam() throws IOException
...
+                .body("name", equalTo(tenant))
+                .body("screen", equalTo(LOGIN_SCREEN))
+                .body("locale", equalTo(DEFAULT_LOCALE));
+
+        JSONObject expectedPreference = new JSONObject(new JSONObject(readResource("add-custom-text.json")).
+                get("preference").toString());
+        JSONObject receivedPreference = new JSONObject(new JSONObject(response.asString()).
+                get("preference").toString());
+        Assert.assertTrue(areJSONObjectsEqual(expectedPreference, receivedPreference),
+                "The custom text preference schema of the Response is incorrect");
```
Comment (ThaminduDilshan) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/rest/api/server/branding/preference/management/v1/BrandingPreferenceManagementSuccessTest.java:363]:
```suggestion
                "The custom text preference schema of the Response is incorrect.");
```

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/rest/api/server/branding/preference/management/v1/BrandingPreferenceManagementSuccessTest.java [hunk 5]
```diff
@@ -318,4 +318,126 @@ public void testDeleteBrandingPreferenceByLocaleQueryParam() throws IOException
...
+    }
+
+    @Test(dependsOnMethods = {"testGetCustomTextPreference"})
+    public void testGetCustomTextPreferenceByQueryParams() throws IOException, JSONException {
```
Comment (ThaminduDilshan) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/rest/api/server/branding/preference/management/v1/BrandingPreferenceManagementSuccessTest.java:367]: Shall we combine this and above method with a data provider?

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/rest/api/server/branding/preference/management/v1/BrandingPreferenceManagementSuccessTest.java [hunk 6]
```diff
@@ -318,4 +318,126 @@ public void testDeleteBrandingPreferenceByLocaleQueryParam() throws IOException
...
+        response.then()
+                .log().ifValidationFails()
+                .assertThat()
+                .statusCode(HttpStatus.SC_OK)
+                .body("type", equalTo(ORGANIZATION_TYPE))
+                .body("name", equalTo(tenant))
+                .body("screen", equalTo(LOGIN_SCREEN))
+                .body("locale", equalTo(DEFAULT_LOCALE));
+
+        JSONObject expectedPreference = new JSONObject(new JSONObject(readResource("update-custom-text.json")).
```
Comment (ThaminduDilshan) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/rest/api/server/branding/preference/management/v1/BrandingPreferenceManagementSuccessTest.java:403]:
```suggestion
        JSONObject expectedPreference = new JSONObject(new JSONObject(body).
```

Comment (UdeshAthukorala) [reply modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/rest/api/server/branding/preference/management/v1/BrandingPreferenceManagementSuccessTest.java]: Fixed with https://github.com/wso2/product-is/pull/16854/commits/f4920b1c7614e08eb8ca0aa4b15907e0c83be572

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/rest/api/server/branding/preference/management/v1/BrandingPreferenceManagementSuccessTest.java (hunk 3 above)
Comment (UdeshAthukorala) [reply modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/rest/api/server/branding/preference/management/v1/BrandingPreferenceManagementSuccessTest.java]: Fixed with https://github.com/wso2/product-is/pull/16854/commits/f4920b1c7614e08eb8ca0aa4b15907e0c83be572

**Review by ThaminduDilshan — APPROVED** [review]

**Review by ThaminduDilshan — APPROVED** [review]

**Pull request merged**